pipeline/src/node_properties/usb_camera_properties.rs

```diff
@@ -12,8 +11,9 @@ pub struct UsbCameraProperties {
     pub runtime: UsbCameraRuntime,
 }
 
-#[derive(Debug, Default, Clone, PartialEq, Serialize, Deserialize)]
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
 pub struct UsbCameraRuntime {
+    pub uri: url::Url,
     #[serde(skip_serializing_if = "Option::is_none")]
     pub source: Option<String>,
```

Comment:
Hm, `source` should be in parent struct, otherwise we cannot get `uri`.

Reply:
> Hm, source should be in parent struct

Ok.

> otherwise we cannot get uri.

Didn't get this part. How is this related?

Reply:
`source` is intended to be passed to us by user, and then we use `source` to query the camera and its `uri`.
Actually I just forgot that `runtime` is flattened, so we still can get `uri`, never mind :)

Reply:
> Actually I just forgot that `runtime` is flattened, so we still can get `uri`, never mind :)

Even if it wasn't, i don't see why it would be an issue since `runtime` is not an optional field.

> `source` is intended to be passed to us by user, and then we use `source` to query the camera and its `uri`.

Right so then it's not a deployment/runtime property and should be moved to main struct after all? Same for other cameras it seems. Also, not optional then, right?

Reply:
> Even if it wasn't, i don't see why it would be an issue since `runtime` is not an optional field.
If it is not flattened and user POSTs a JSON without `runtime` then it won't deserialize.

> Also, not optional then, right?
Yeah, `source` is required at the moment of deployment (I think the first time we deserialize this struct is at deployment).

Reply:
Thanks. So a required prop but still in deployment. Gotcha!

Reply:
Not fixing this one in a `fixme` commit cause the rebase would be a bit of nightmare so additional patch on top is hopefully fine.